controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,906 +45,847 @@ import (
+
+		// Calculate how many Machines have health check succeeded = false.
+		g.Eventually(func() int {
+			machines := &clusterv1.MachineList{}
+			err := testEnv.List(ctx, machines, client.MatchingLabels{
+				"selector": mhc.Spec.Selector.MatchLabels["selector"],
+			})
+			if err != nil {
+				return -1
 			}
-			machineSet.Default()
-			Expect(testEnv.Create(ctx, machineSet)).To(Succeed())
-
-			// Calculate how many Machines have health check succeeded = false.
-			Eventually(func() int {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
-				return len(machines.Items)
-			}).Should(Equal(1))
-
-			// Create the MachineHealthCheck instance.
-			mhc.Spec.ClusterName = cluster.Name
-			mhc.Spec.NodeStartupTimeout = &metav1.Duration{Duration: 1 * time.Second}
-			mhc.SetGenerateName("unhealthy-machines-deleted-")
-			Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-
-			// Pause the MachineSet reconciler.
-			machineSetPatch := client.MergeFrom(machineSet.DeepCopy())
-			machineSet.Annotations = map[string]string{
-				clusterv1.PausedAnnotation: "",
-			}
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Calculate how many Machines should be remediated.
-			var unhealthyMachine *clusterv1.Machine
-			Eventually(func() (remediated int) {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
-
-				for i := range machines.Items {
-					if conditions.Get(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) != nil {
-						unhealthyMachine = machines.Items[i].DeepCopy()
-						remediated++
-					}
-				}
-				return
-			}).Should(Equal(1))
-
-			// Unpause the MachineSet reconciler.
-			machineSetPatch = client.MergeFrom(machineSet.DeepCopy())
-			delete(machineSet.Annotations, clusterv1.PausedAnnotation)
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Make sure the Machine gets deleted.
-			Eventually(func() bool {
-				machine := unhealthyMachine.DeepCopy()
-				err := testEnv.Get(ctx, util.ObjectKey(unhealthyMachine), machine)
-				return apierrors.IsNotFound(err) || !machine.DeletionTimestamp.IsZero()
+			return len(machines.Items)
+		}, timeout, time.Second).Should(Equal(1))
+
+		// Create the MachineHealthCheck instance.
+		mhc.Spec.NodeStartupTimeout = &metav1.Duration{Duration: time.Second}
+
+		g.Expect(testEnv.Create(ctx, mhc)).To(Succeed())
+		// defer cleanup for all the objects that have been created
+		defer func(do ...runtime.Object) {
+			g.Expect(testEnv.Cleanup(ctx, do...)).To(Succeed())
+		}(cluster, mhc, infraTmpl, machineSet)
+
+		// Pause the MachineSet reconciler. WHY?? I SEE WHAT IS HAPPENING BUT
```

Comment:
This is to delay the deletion of the Machine, which the MachineSet controller does when it sees a Machine marked unhealthy by MHC.

Reply:
Whoa! Thought I removed all of my ranting comments. Thanks for catching this. 😬